### DISCUSSION TITLE
Fire onChange handler?

### DISCUSSION THREAD

**Discussion by joenewstrom** [drivardxrm/LookupDropdown.PCF#8] [Ideas]
David, is there a good reason that the control should not fire the onChange Handler for the control? I have a workaround currently by adding a second hidden control for the same attribute, but if onChange were fired (or maybe a boolean that would allow you to turn it on and off?) that would be a nice feature.

**Comment by drivardxrm** [comment]
Hi @joenewstrom,

Interesting, 
I never noticed that the OnChange handler was not triggered.
I thought this would be handled natively by the PCF platform.

I will investigate on my side and I will try to make it fire directly from the control as it should be.

Thanks for the feedback, it's really appreciated.

**Comment by drivardxrm** [comment]
Hi @joenewstrom,

It seems to work correctly on my side.

Just to be sure I understand your issue correctly :

1- You have a lookup on your form with the lookup dropdown pcf installed and configured
2- You have a JS file registered on the form with an event handler configured on the lookup field OnChange event
3- When you change the value of the lookup field via the PCF control, nothing happens

Am I right ?

I did a very simple example on my test environment and I get expected results
When I change the value of the lookup via the PCF the JS function gets called
